bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -78,16 +92,27 @@ async def list_watched_users(
         The optional kwarg `update_cache` specifies whether the cache should
         be refreshed by polling the API.
         """
-        # TODO Once the watch channel is removed, this can be done in a smarter way, without splitting and overriding
-        # the list_watched_users function.
-        watched_data = await self.prepare_watched_users_data(ctx, oldest_first, update_cache)
+        successful_update = False
+        if update_cache:
+            if not (successful_update := await self.refresh_cache()):
+                await ctx.send(":warning: Unable to update cache. Data may be inaccurate.")
 
-        if update_cache and not watched_data["updated"]:
-            await ctx.send(f":x: Failed to update {self.__class__.__name__} user cache, serving from cache")
+        nominations = self.cache.items()
+        if oldest_first:
+            nominations = reversed(nominations)
 
         lines = []
-        for user_id, line in watched_data["info"].items():
-            if self.watched_users[user_id]['reviewed']:
+
+        for user_id, user_data in nominations:
+            member = ctx.guild.get_member(user_id)
+            line = f"• `{user_id}`"
+            if member:
+                line += f" ({member.name}#{member.discriminator})"
+            inserted_at = user_data['inserted_at']
+            line += f", added {get_time_delta(inserted_at)}"
+            if not member:  # Cross off users who left the server.
+                line = f"~~{line}~~"
+            if user_data['reviewed']:
```

Comment:
This exact code exists in the watch channels code. Can you refactor this into a utility function that both can use? Not sure what the best location for it would be.